src/IonTextReader.ts

```diff
@@ -192,31 +192,29 @@ next() {
     return this._parser.isNull();
   }
 
-  stringValue() : string {
-    var i, s, t = this;
-    this.load_raw();
-    if (t.isNull()) {
-      s = "null";
-      if (t._type != IonTypes.NULL) {
-        s += "." + t._type.name;
-      }
-    }
-    else if (t._type.scalar) {
-      // BLOB is a scalar by you don't want to just use the string 
-      // value otherwise all other scalars are fine as is
-      if (t._type !== IonTypes.BLOB) {
-        s = t._raw;
-      }
-      else {
-        s = t._raw;   // TODO - this is a temp fix !!
-      }
-    }
-    else {
-      i = t.ionValue();
-      s = i.stringValue();
+    stringValue() : string {
+        this.load_raw();
+        if (this.isNull()) return (this._type === IonTypes.NULL) ? "null" : "null." + this._type.name;
+        if (this._type.scalar) {
```

Comment:
Should it be an error otherwise?

Reply:
A number of the types have no representation outside of string/text/Number. Only the more complex ones like IonDecimal have real OO and functionality tied to them.

Reply:
yes